pkg/controller/deployment/deployment_controller.go

```diff
@@ -609,7 +620,11 @@ func (dc *DeploymentController) scaleRC(rc *api.ReplicationController, newScale
 	return dc.client.ReplicationControllers(rc.ObjectMeta.Namespace).Update(rc)
 }
 
-func (dc *DeploymentController) updateDeployment(deployment *extensions.Deployment) (*extensions.Deployment, error) {
-	// TODO: Using client for now, update to use store when it is ready.
-	return dc.expClient.Deployments(deployment.ObjectMeta.Namespace).Update(deployment)
+// hasTimedOut returns if the provided deployment has timed-out.
+func (dc *DeploymentController) hasTimedOut(d extensions.Deployment, latestState api.ReplicationController) bool {
+	if d.Spec.MaxProgressThresholdSeconds == nil || latestState.CreationTimestamp.IsZero() {
+		return false
+	}
+	maxDelta := time.Duration(*d.Spec.MaxProgressThresholdSeconds) * time.Second
+	return latestState.CreationTimestamp.Add(maxDelta).Before(time.Now())
```

Comment:
I'm not sure that this accurately captures whether progress has been made in a timely fashion since the last sync. Seems like you'd need to have awareness of the time either the old RC(s) or new RC was last scaled so you can decide whether it has been `MaxProgressThresholdSeconds` since the last successful scaling event.